domain/src/test/java/com/thoughtworks/go/config/materials/git/GitMaterialTest.java

```diff
@@ -749,6 +747,26 @@ void shouldCopyOverSecretResolvedParams() {
         }
     }
 
+    @Test
+    void populateEnvContextShouldSetMaterialEnvVars() {
+        GitMaterial material = new GitMaterial("https://user:password@example.github.com");
```

Comment:
Same comment as above, use a path and an explicit branch, rather than default.

Reply:
Refactored to include the suggested changes. Added a separate test to check for default value being set.